### PR TITLE
Remove links to pure-docker

### PR DESCRIPTION
**Pull request by pecigonzalo**
`pure-docker` repository is just a reference and not a supported deployment type, this aims to clarify that.

### CODE REVIEW

File: doc/admin/install/index.md [hunk 1]
```diff
@@ -2,10 +2,15 @@
...
+| [Single-container server](../install/docker/index.md) | Local testing                                       | 60 seconds | No             | No            | No          |
+| [Docker Compose](../install/docker-compose/index.md)  | Small & medium production deployments               | 5 minutes  | Not Supported  | No            | Yes         |
+| [Kubernetes](../install/kubernetes/index.md)          | Medium & large highly-available cluster deployments | 30 minutes | Yes            | Yes           | Yes         |
 
-* If you're just starting out, we recommend [running Sourcegraph locally](docker/index.md). It takes only a few minutes and lets you try out all of the features.
 
+* If you're just starting out, we recommend [running Sourcegraph locally](docker/index.md). It takes only a few minutes and lets you try out all of the features.
 * If you need scalability and high-availability beyond what a single-node [Docker Compose](https://docs.docker.com/compose/) can offer, use the [Kubernetes cluster deployment option](https://github.com/sourcegraph/deploy-sourcegraph), instead.
+* Other deployment types are **not supported**.
```
Comment (slimsag) [inline doc/admin/install/index.md:12]:
```suggestion
```

I do not think we need to state this, we just need to state was is supported.

File: doc/admin/install/cluster.md [hunk 2]
```diff
@@ -1,15 +0,0 @@
-# Installing Sourcegraph on a cluster
```
Comment (slimsag) [inline doc/admin/install/cluster.md:1]: Please add a redirect for this page for Google

Comment (pecigonzalo) [reply doc/admin/install/cluster.md:1]: :+1: I dont know how this is done in our docs, do you have an example?

Comment (slimsag) [reply doc/admin/install/cluster.md:1]: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@gp/dyi-deployment/-/blob/doc/_resources/assets/redirects

Once updated the docsite pod will need to be rebooted

Comment (slimsag) [reply doc/admin/install/cluster.md:1]: Would be great to document this somewhere, I had to go digging for it

**Comment by slimsag** [comment]
I think the linter is complaining because the sidebar is linking to the cluster doc on every page: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24%40gp/dyi-deployment+file:doc+/cluster&patternType=literal

**Comment by slimsag** [comment]
Unfortunately in some cases like this the linter can be SUPER unclear

File: doc/admin/install/cluster.md (hunk 2 above)
Comment (pecigonzalo) [reply doc/admin/install/cluster.md:1]: Done

**Review by slimsag — APPROVED** [review]

**Pull request merged**